gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/cache/ApiKeyCacheManager.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.handlers.api.cache;
+
+import io.gravitee.gateway.api.cache.ApiKey;
+import io.gravitee.node.api.cache.Cache;
+import io.gravitee.node.api.cache.CacheManager;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ApiKeyCacheManager implements io.gravitee.gateway.api.cache.ApiKeyCacheManager {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ApiKeyCacheManager.class);
+    private static final String CACHE_NAME = "API_KEYS";
+
+    private final Cache<String, ApiKey> cache;
+
+    public ApiKeyCacheManager(CacheManager cacheManager) {
+        cache = cacheManager.getOrCreateCache(CACHE_NAME);
+    }
+
+    @Override
+    public void save(ApiKey apiKey) {
+        String cacheKey = buildCacheKey(apiKey);
+        if (apiKey.isRevoked() || apiKey.isPaused()) {
+            LOGGER.debug(
+                "Remove an api-key from cache [id: {}] [plan: {}] [app: {}]",
+                apiKey.getId(),
+                apiKey.getPlan(),
+                apiKey.getApplication()
+            );
+            cache.evict(cacheKey);
+        } else {
+            LOGGER.debug("Caching an api-key [id: {}] [plan: {}] [app: {}]", apiKey.getId(), apiKey.getPlan(), apiKey.getApplication());
+            cache.put(cacheKey, apiKey);
+        }
+    }
+
+    @Override
+    public Optional<ApiKey> get(String api, String key) {
+        return Optional.ofNullable(cache.get(buildCacheKey(api, key)));
+    }
+
+    private String buildCacheKey(ApiKey apiKey) {
+        return buildCacheKey(apiKey.getApi(), apiKey.getKey());
+    }
+
+    private String buildCacheKey(String api, String key) {
+        return String.format("%s.%s", api, key);
```

Comment:
I would have align the cache keys constructions with the subscriptions

```suggestion
        return api + '-' + key;
```

Reply:
Nice catch.
I did the contrary : construct the subscription key the same way as api keys, with the ".";
I prefer the "." as separator cause uuid already contains "-", it's visually simpler to read.